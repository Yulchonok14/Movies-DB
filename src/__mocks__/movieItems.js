export const movieItems = [
    {
        id: 111,
        title: 'Bloodshot',
        date: '2020',
        rating: '3.6',
        genre: 'Action, Drama, Sci-Fi',
        imageUrl: 'https://upload.wikimedia.org/wikipedia/ru/9/9d/Bloodshot_poster.jpg',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '110'
    },
    {
        id: 222,
        title: 'The invisible man',
        date: '2019',
        rating: '4.6',
        genre: 'Horror, Sci-Fi, Thriller',
        imageUrl: 'https://avatars.mds.yandex.net/get-kinopoisk-image/1900788/a90dd0dd-a8ae-4112-9865-23eb8f5879a8/360',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '165'
    },
    {
        id: 333,
        title: 'Monster trucks',
        date: '2019',
        rating: '3.9',
        genre: 'Adventure, Action, Comedy',
        imageUrl: 'https://upload.wikimedia.org/wikipedia/ru/5/5e/Monster_Trucks_%28film%29.jpg',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '154'
    },
    {
        id: 444,
        title: 'Cold comes the night',
        date: '2016',
        rating: '4.4',
        genre: 'Drama, Thriller, Crime',
        imageUrl: 'https://laughingsquid.com/wp-content/uploads/2014/01/CCTN_poster_Billy_HD-for-web.jpg',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '129'
    },
    {
        id: 555,
        title: 'Snowden',
        date: '2010',
        rating: '4.8',
        genre: 'Drama, Thriller, Biography',
        imageUrl: 'https://upload.wikimedia.org/wikipedia/ru/1/1a/Snowden_%28film%29.jpg',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '113'
    },
    {
        id: 666,
        title: 'Knock knock',
        date: '2011',
        rating: '2.6',
        genre: 'Horror, Thriller',
        imageUrl: 'https://i1.wp.com/morbidlybeautiful.com/wp-content/uploads/2015/10/kk-2.jpg',
        description: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        duration: '187'
    }
];