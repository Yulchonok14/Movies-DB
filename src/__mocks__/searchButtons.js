export const searchByButtonsArr = [
    {id: 111, text: 'title', isActive: true},
    {id: 222, text: 'genre', isActive: false}
];