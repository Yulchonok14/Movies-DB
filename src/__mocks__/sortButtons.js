export const sortByButtonsArr = [
    {id: 111, text: 'release date', isActive: true},
    {id: 222, text: 'rating', isActive: false}
];