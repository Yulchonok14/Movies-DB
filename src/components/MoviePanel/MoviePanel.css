.MoviePanel {
    display: flex;
    justify-content: stretch;
    padding: 0 20px 50px;
    color: #FFFFFF;
    flex-direction: column;
    align-items: center;
}

.MoviePanel img {
    width: 70vw;
}

.Details {
    flex-grow: 1;
}

.Details label {
    font-size: 13px;
    opacity: 0.5;
    padding: 0 8px;
    margin-right: 40px;
}

.Details div {
    display: flex;
    align-items: flex-end;
    margin: 40px 0;
}

.Title {
    font-size: 40px;
    opacity: 0.8;
}

.Rating {
    color: #A1E66F;
    font-size: 25px;
    margin-left: 25px;
    border: 1px solid rgba(255,255,255,0.5);
    padding: 15px;
    border-radius: 35px;
    width: 35px;
    text-align: center;
}

.Date {
    color: #F65261;
    font-size: 25px;
}

.Duration {
    color: #F65261;
    font-size: 25px;
}

.Description {
    line-height: 24px;
    max-height: 200px;
    overflow: auto;
}

@media (min-width: 600px) and (max-width: 1000px) {
    .MovieItem {
        width: 35vw;
    }

    .MoviePanel {
        padding: 0 20px 50px;
        flex-direction: row;
    }

    .Details {
        width: 70vw;
        padding-left: 50px;
    }

    .Details div:first-of-type {
        display: flex;
        align-items: flex-start;
    }

    .Description {
        max-height: 125px;
    }

    .MoviePanel img {
        height: 35vh;
        width: auto;
    }
}

@media (min-width: 1001px) {
    .Title {
        font-size: 52px;
    }

    .MoviePanel {
        padding: 0 100px 50px;
        flex-direction: row;
    }

    .Details {
        width: 70vw;
        padding-left: 50px;
    }

    .Details div:first-of-type {
        display: flex;
        align-items: flex-start;
    }

    .MoviePanel img {
        height: 45vh;
        width: auto;
    }
}