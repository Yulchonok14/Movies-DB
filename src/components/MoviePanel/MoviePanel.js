import React from 'react';
import PropTypes from 'prop-types';

import classes from './MoviePanel.css';

const MoviePanel = (props) => (
    <div className={classes.MoviePanel}>
        <img src={props.selectedMovie.imageUrl} alt="Selected movie poster."/>
        <div className={classes.Details}>
            <div>
                <span className={classes.Title}>{props.selectedMovie.title}</span>
                <span className={classes.Rating}>{props.selectedMovie.rating}</span>
            </div>
            <div>
                <span className={classes.Date}>{props.selectedMovie.date}</span><label>year</label>
                <span className={classes.Duration}>{props.selectedMovie.duration}</span><label>min</label>
            </div>
            <p className={classes.Description}>{props.selectedMovie.description}</p>
        </div>
    </div>
);

MoviePanel.propTypes = {
    selectedMovie: PropTypes.shape({
        id: PropTypes.number.isRequired,
        title: PropTypes.string.isRequired,
        rating: PropTypes.string.isRequired,
        imageUrl: PropTypes.string.isRequired,
        genre: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired,
        description: PropTypes.string.isRequired,
        duration: PropTypes.string.isRequired
    })
};

export default MoviePanel;