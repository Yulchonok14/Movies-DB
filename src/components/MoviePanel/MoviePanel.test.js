import React from 'react';

import { configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import MoviePanel from './MoviePanel';
import { movieItems } from '../../__mocks__/movieItems';

configure({
    adapter: new Adapter()
});

describe('<MoviePanel/>', () => {

    it('MoviePanel renders correctly', () => {
        const tree = renderer
            .create(<MoviePanel selectedMovie={movieItems[5]}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});