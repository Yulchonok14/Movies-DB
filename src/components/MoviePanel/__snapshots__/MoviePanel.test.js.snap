// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviePanel/> MoviePanel renders correctly 1`] = `
<div>
  <img
    alt="Selected movie poster."
    src="https://i1.wp.com/morbidlybeautiful.com/wp-content/uploads/2015/10/kk-2.jpg"
  />
  <div>
    <div>
      <span>
        Knock knock
      </span>
      <span>
        2.6
      </span>
    </div>
    <div>
      <span>
        2011
      </span>
      <label>
        year
      </label>
      <span>
        187
      </span>
      <label>
        min
      </label>
    </div>
    <p>
      Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”
    </p>
  </div>
</div>
`;
